exotations/exotica_core_task_maps/src/manipulability.cpp

```diff
@@ -0,0 +1,65 @@
+//
+// Copyright (c) 2018, University of Edinburgh
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//  * Redistributions of source code must retain the above copyright notice,
+//    this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of  nor the names of its contributors may be used to
+//    endorse or promote products derived from this software without specific
+//    prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+
+#include <exotica_core_task_maps/manipulability.h>
+
+REGISTER_TASKMAP_TYPE("Manipulability", exotica::Manipulability);
+
+namespace exotica
+{
+Manipulability::Manipulability() = default;
+Manipulability::~Manipulability() = default;
+
+double Manipulability::get_manipulability_index(int& i)
+{
+    if (i < 0 || i >= n_end_effs_) ThrowNamed("Value for i is out of range!");
+    const Eigen::MatrixXd& J = kinematics[0].jacobian[i].data;
+    return std::sqrt((J * J.transpose()).determinant());
+}
+
+void Manipulability::Update(Eigen::VectorXdRefConst x, Eigen::VectorXdRef phi)
+{
+    if (phi.rows() != TaskSpaceDim()) ThrowNamed("Wrong size of phi!");
+
+    for (int i = 0; i < n_end_effs_; ++i)
+    {
+        phi(i) = get_manipulability_index(i);
```

Comment:
Do you need the exposed-to-Python? You can get the `Phi` of the taskmap... and I'd rather inline the function above than have this call here.

Reply:
I thought it would be useful for users to have the facility to get the manipulability index in python, perhaps for logging or further analysis purposes.

Reply:
They can via the `Phi`: you can get the start and length of the task map and thus extract it there. It _may_ be but it's limited to Python only (these calls do not work in C++, cf. #437) and I'd like the declaration of the function to be `inline` if we keep it here.

Reply:
Ok, will revert changes